designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessActionRepository.scala

```diff
@@ -163,4 +166,44 @@ extends Repository[F] with EspTables with CommentActions with ProcessActionRepos
     run(processActionsTable.filter(_.state === ProcessActionState.InProgress).delete.map(_ => ()))
   }
 
+  override def getLastFinishedActionPerProcess(actions: Option[List[ProcessActionType]]): F[Map[ProcessId, ProcessAction]] = {
+    val query = processActionsTable
+      .filter(_.state === ProcessActionState.Finished)
+      .groupBy(_.processId)
+      .map { case (processId, group) => (processId, group.map(_.performedAt).max) }
+      .join(processActionsTable)
+      .on { case ((processId, maxPerformedAt), action) => action.processId === processId && action.state === ProcessActionState.Finished && action.performedAt === maxPerformedAt } //We fetch exactly this one  with max deployment
+      .map { case ((processId, _), action) => processId -> action }
+      .joinLeft(commentsTable)
+      .on { case ((_, action), comment) => action.commentId === comment.id }
+      .map { case ((processId, action), comment) => processId -> (action, comment) }
+
+    run(actions
+      .map(actions => query.filter { case (_, (entity, _)) => entity.action.inSet(actions) })
+      .getOrElse(query)
+      .result.map(_.toMap.mapValuesNow(toProcessAction)))
+  }
+
+  override def getProcessActions(processId: ProcessId)(implicit ec: ExecutionContext): F[List[ProcessAction]] =
```

Comment:
getProcessFinishedActions? To be consistent with this below? Here you also check `ProcessActionState.Finished`.

Reply:
done